src/core/mac/sub_mac.cpp

```diff
@@ -261,14 +262,21 @@ void SubMac::HandleReceiveDone(RxFrame *aFrame, Error aError)
         UpdateFrameCounter(aFrame->mInfo.mRxInfo.mAckFrameCounter);
     }
 
-#if OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
+#if OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE || OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
     if (aFrame != nullptr && aError == kErrorNone)
     {
+        if ((mState == kStateCslSample) && aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck)
+        {
+            mCslLastSync = TimerMicro::GetNow();
```

Comment:
How can we assume an Enh-ACK is really received by peer, thus making the assumption that CSL transmitter has been synchronized?

Reply:
You are right, however I believe the risk of losing the Enh-ACK is pretty low compared to the power consumption advantage of keeping the receive window small. Specially taking into account that an SSED is likely to not send any frame to the parent apart from ACKs.

Reply:
Then I think it would be better to add some comments here.

Another thing is, I don't think this check is needed `mState == kStateCslSample` as OT stack might have already change to sleep state when the TxDone even is called. As long as the transmitted frame is acked with CSL IE included, we can consider CSL transmitter being synchronized. How do you think?

Reply:
Maybe we should just check that `mCslPeriod > 0.`